### ISSUE TITLE
Headerbars do not respond to touchscreen events

### ISSUE THREAD

**Issue by spl237** [labwc/labwc#1545]
Applications themselves respond to touchscreen events, but it looks as if the labwc headerbars do not have touch handling enabled.

If you run labwc on a touchscreen, then you can interact with the contents of windows, scrollbars etc. But it is not possible to activate the minimise / maximise or close buttons, or to drag the headerbar to move the window.

Has this not been implemented, or is there something I need to enable?

**Comment by Consolatis** [comment]
We might not do mouse emulation for touch events, I don't have hardware to test this but `src/input/touch.c` is pretty small.
It seems to indeed completely skip non client surfaces, maybe we could emulate usual cursor events in those cases instead. Similar to `src/input/tablet.c` (e.g. via `cursor_emulate_move_absolute()` and `cursor_emulate_button()` after a timeout on `touch_down()`).

**Comment by spl237** [comment]
OK, I have written a very crude bodge using cursor_emulate functions which means I can operate the close, minimize etc buttons on a headerbar with the touchscreen. I haven't yet got window dragging working, but this seems to at least be a viable solution in the short term - I'll hack about and see what I can get working.

(But if someone else knows more about this than me - which is almost certainly the case - do please feel free to come up with something better!)

**Comment by spl237** [comment]
Right, I have window dragging and button clicking working, but the problem I now have is that this breaks behaviour in a few other areas of the system. I think what I need to do is to ensure that the cursor_emulate functions are only applied when a touch is on an SSD titlebar.

Is there any easy way to determine if the item under the touch point is an SSD titlebar? If I could do that, I think this would work pretty much perfectly...

**Comment by spl237** [comment]
It looks as if just checking if the surface returned from touch_get_coords is valid can be used to check whether you are on the SSD or not, as because the SSD seems to be the only thing that is returned as not a valid touch surface. That assumption may of course be incorrect!
